### PR TITLE
API app built

### PR DESCRIPTION
**Pull request by Lordkaito**
Our team consists of four members, @see-why, @leolpaz, @redwing555, and @Lordkaito.
This is the PR for the back-end side of our project.

In this branch: 
1. Created a new API-only application using Ruby on Rails.
2. Created migrations for the database to include the necessary tables.
3. Created the necessary models (cars and reservations) for the app to run properly.
4. Created different endpoints to the API so the front-end side can do different actions.
5. Created the necessary routes to manage the API properly.
6. Test the app using RSpec-rails and rswag.
7. Included basic instructions into the Readme file to let the user know the basic flow of the API.

The link to the front-end PR is [here](https://github.com/redwing555/final-capstone-frontend/pull/37)

### CODE REVIEW

File: app/controllers/api/v1/reservations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  def index
+    @user_name = params[:username]
+
+    if @user_name
+      @reservations = Reservation.includes(:car).where(username: @user_name)
+      render json: @reservations
+    else
+      @reservation = Reservation.all
+      render json: @reservation
```
Comment (MuriloRoque) [inline app/controllers/api/v1/reservations_controller.rb]: - [ ] I wonder if you misspelled this one and forgot about the `s`, shouldn't it be `@reservations`?

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,10 @@
...
-
-  # Defines the root path route ("/")
-  # root "articles#index"
-end
+  mount Rswag::Ui::Engine => '/api-docs'
+  mount Rswag::Api::Engine => '/api-docs'
+  namespace :api do
+    namespace :v1 do
+      resources :cars
+      resources :reservations
```
Comment (MuriloRoque) [inline config/routes.rb]: - [ ] I suggest that you specify the routes you will really be using here so that you don't create any unnecessary ones.

**Review by MuriloRoque — CHANGES_REQUESTED** [review]
Hi @Lordkaito, @see-why, @leolpaz, @redwing555,

Very good job so far!
There are only some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better.

## Highlights

- Very professional repository.
- It's nice that you built tests, even though it isn't in the requirements.
- Your code is clean, I really liked it.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/controllers/api/v1/reservations_controller.rb (hunk 1 above)
Comment (see-why) [reply app/controllers/api/v1/reservations_controller.rb]: thanks for pointing this out, @reservations is a more meaningful variable name

File: config/routes.rb (hunk 2 above)
Comment (see-why) [reply config/routes.rb]: Thanks, we will do that.

**Review by MuriloRoque — APPROVED** [review]
Hi @Lordkaito, @see-why, @leolpaz, @redwing555,

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**